aruco_opencv/src/aruco_tracker.cpp

```diff
@@ -408,6 +427,15 @@ class ArucoTracker : public rclcpp_lifecycle::LifecycleNode
     }
   }
 
+  void callback_compressed_image(const sensor_msgs::msg::CompressedImage::ConstSharedPtr img_msg)
+  {
+    // Decompress the image using bridge
+    auto image = cv_bridge::toCvCopy(img_msg, "bgr8");
+    auto new_msg = image->toImageMsg();
```

Comment:
`toImageMsg()` method creates another copy of the image data. If possible, I would like to avoid doing any unnecessary copies.

Reply:
Unfortunately it seems that OpenCV does not allow to imdecode into an existing buffer which complicates things a lot. It makes it virtually impossible to create a new sensor_msgs/Image without copying. I could split `callback_image` into two parts where one handles the conversion to CvImage and then calls the other part to run detection. Then both `callback_image` and `callback_compressed_image` would use this method to detect markers.

Reply:
Let me know if I should continue to work in this direction.

Reply:
Yes, this would be a preferred method.

Reply:
It's done.